File: solutions/php/0167-two-sum-ii-input-array-is-sorted.php
<?php

class Solution
{
    /**
     * @param int[] $numbers
     * @param int $target
     * @return int[]
     */
    public function twoSum(array $numbers, int $target): array
    {
        $pairs = [];

        foreach ($numbers as $key => $value) {
            if (array_key_exists($target - $value, $pairs)) {
                return [$pairs[$target - $value], $key + 1];
            }

            $pairs[$value] = $key + 1;
        }

        return [];
    }
}

File: solutions/go/0167-two-sum-ii-input-array-is-sorted.go
package main

func twoSum(numbers []int, target int) []int {
	pairs := make(map[int]int)

	for key, value := range numbers {
		if pair, ok := pairs[target-value]; ok {
			return []int{pair, key + 1}
		}

		pairs[value] = key + 1
	}

	return nil
}
